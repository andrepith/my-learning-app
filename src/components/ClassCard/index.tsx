import React from 'react';
import { Link } from 'react-router-dom';
import { ClassesInterface } from '../../utils/constant';

function ClassCard({
  detail = false,
  data: { id, name, mentors, description },
}: ClassesInterface) {
  return (
    <div className="class-card">
      <h3>
        Kelas-
        {id}
      </h3>
      <p>{detail ? description : name}</p>
      {!detail && (
        <button type="button">
          <Link to={`/detail/${id}`}>Details</Link>
        </button>
      )}
      {detail &&
        mentors?.map((item) => (
          <div>
            <h3>
              Mentor-{item.id}-{item.name}
            </h3>
            <p>{item.description}</p>
          </div>
        ))}
    </div>
  );
}

export default ClassCard;
