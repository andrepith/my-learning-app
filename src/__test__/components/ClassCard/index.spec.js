import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import ClassCard from '../../../components/ClassCard';

const homeProps = {
  id: 1,
  name: '',
};

const detailProps = {
  id: 1,
  name: '',
  mentors: [
    {
      id: 1,
      name: '',
      description: '',
    },
  ],
  description: '',
};

it('Renders on Home', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <ClassCard data={homeProps} />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders on Detail', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <ClassCard detail data={detailProps} />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
