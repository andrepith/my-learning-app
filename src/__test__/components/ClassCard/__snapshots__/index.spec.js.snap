// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders match snapshot on Detail 1`] = `
<div
  className="class-card"
>
  <h3>
    Kelas-
    1
  </h3>
  <p>
    
  </p>
  <div>
    <h3>
      Mentor-
      1
      -
      
    </h3>
    <p>
      
    </p>
  </div>
</div>
`;

exports[`Renders match snapshot on Home 1`] = `
<div
  className="class-card"
>
  <h3>
    Kelas-
    1
  </h3>
  <p>
    
  </p>
  <button
    type="button"
  >
    <a
      href="/detail/1"
      onClick={[Function]}
    >
      Details
    </a>
  </button>
</div>
`;
